GVFS/GVFS/CommandLine/GVFSVerb.cs

```diff
@@ -113,7 +113,7 @@ public static bool TrySetRequiredGitConfigSettings(Enlistment enlistment)
                 { "core.safecrlf", "false" },
                 { "core.untrackedCache", "false" },
                 { "core.repositoryformatversion", "0" },
-                { "core.filemode", "false" },
+                { "core.filemode", GVFSPlatform.Instance.GitCoreFileMode ? "true" : "false" },

```

Comment:
I am not sure how git actually uses this value.  Is it possible to write a functional test that would fail if this value were not set correctly?

Reply:
> I thought there was still some outstanding things there?

The two outstanding issues I'm aware of are:

1. VFS4G will not update a placeholder if *only* its mode changes during a projection change (#223)
2. If *only* the mode of a file is changed VFS4G is not notified and the file is not added to ModifiedPaths.dat (#457)

So if `core.filemode` controls whether git checks a file's mode at all, I'd think a test like this would catch it:

1. Modify file
2. git checkout -- <file>
3. Confirm git status is clean
4. chmod file
5. Confirm git status reports file

Reply:
We might want to check that the file mode of a file on disk is set correctly after a checkout. I think I noticed some issues with fast fetch where the execute bit was not set correctly after checking out files.